web/server.js

```diff
@@ -291,6 +292,75 @@ module.exports = function(options) {
         redirectObj.query.logout = true;
         reply.redirect(url.format(redirectObj));
       }
+    },
+    {
+      method: 'GET',
+      path: '/user',
+      config: {
+        auth: false,
+        pre: [
+          {
+            assign: 'requestToken',
+            method: function(request, reply) {
+              var tokenHeader = request.headers.authorization || '';
+              tokenHeader = tokenHeader.split(' ');
+
+              if ( tokenHeader[0] !== 'token' || !tokenHeader[1] ) {
+                return reply(Boom.unauthorized('Missing or invalid authorization header'));
+              }
+
+              reply(tokenHeader[1]);
+            }
+          },
+          {
+            assign: 'token',
+            method: function(request, reply) {
+              oauthDb.lookupAccessToken(request.pre.requestToken, function(err, token) {
+                if ( err ) {
+                  return reply(err);
+                }
+
+                if ( token.expires_at <= Date.now() ) {
+                  return reply(Boom.unauthorized('Expired token'));
+                }
+
+                var tokenScopes = token.scopes.split(' ');
+
+                if ( tokenScopes.indexOf('user') === -1 && tokenScopes.indexOf('email') === -1 ) {
+                  reply(Boom.unauthorized('The token does not have the required scopes'));
+                }
+
+                reply(token);
+              });
+            }
+          },
+          {
+            assign: 'user',
+            method: function(request, reply) {
+              account.getUser(request.pre.token.user_id, function(err, json) {
```

Comment:
I *think* that `request.pre.token.user_id` is `undefined` here because I added some logging to `getUser` and it thinks its first argument is undefined. Which means that it's ultimately hitting `https://login.webmaker.org/user/username/undefined`, which would explain the weird [404 I'm getting](https://github.com/mozilla/id.webmaker.org/pull/128#issuecomment-90271655).